File: backend/openai_helper.mjs
import OpenAI from 'openai';
import dotenv from 'dotenv';

dotenv.config();
const openai = new OpenAI(process.env.OPENAI_API_KEY);

export default async function generate(prompt) {
    const imageCompletion = await openai.images.generate({ prompt: prompt, model: "dall-e-2", size: "256x256" });
    return imageCompletion.data[0]["url"];
}

File: backend/openai_helper.py
"""
API methods for OpenAI
"""

from openai import OpenAI
from dotenv import load_dotenv
import os
import asyncio

load_dotenv()
client = OpenAI(
    api_key=os.getenv("OPENAI_API_KEY")
)


def create_image(prompt: str):
    model = "dall-e-3"
    response = client.images.generate(
        model=model,
        prompt=prompt,
        n=1,
    )

    print(response.data[0]["url"])


if __name__ == "__main__":
    prompt = "A cute cat"
    create_image(prompt)


